lib/auth/auth_with_roles_test.go

```diff
@@ -645,6 +645,107 @@ func TestListNodes(t *testing.T) {
 	require.Empty(t, cmp.Diff(expectedNodes, nodes))
 }
 
+// TestGetAndList_Nodes users can retrieve nodes with various filters
+// and with the appropriate permissions.
+func TestGetAndList_Nodes(t *testing.T) {
```

Comment:
This is mostly a replica of `TestListNodes` (with added filter test) which will become deprecated in favor of `ListResources`.